### ISSUE TITLE
Specify Nuitka version as an input parameter to the Action 

### ISSUE THREAD

**Issue by jimkring** [Nuitka/Nuitka-Action#5]
Users may want to specify the version of nuitka to use for the build. As such, it could be good to allow passing in the nuitka version as an input parameter to the action.

**Comment by kayhayen** [comment]
Maybe that means, one can use the pip installable URL types?

**Comment by jimkring** [comment]
@kayhayen  
  
Yes, my thinking was that a user could specify a \``nuitka-version`\` version like this:

### Semantic Version

```
- Name: My App Build
  uses: Nuitka/Nuitka-Action@v0.5
  with:
    nuitka-version: 1.0
    script-name: my_app.py
    standalone: true
```

### Version URLs

We could (as you suggested) detect a url and have pip install from the url, too. 

```
- Name: My App Build
  uses: Nuitka/Nuitka-Action@v0.5
  with:
    nuitka-version: https://github.com/Nuitka/Nuitka/archive/factory.zip
    script-name: my_app.py
    standalone: true
```

### Special Version Keywords

We could also support some keywords like `factory`.

```
- Name: My App Build
  uses: Nuitka/Nuitka-Action@v0.5
  with:
    nuitka-version: factory
    script-name: my_app.py
    standalone: true
```

### Git Tags/SHA Refs with "@"

We could also detect an `@` symbol before the version and treat that as a git tag/sha (e.g. `@hotfix/0.9.6`) that expands (for pip install) to this  `git+ssh://git@github.com:Nuitka/Nuitka.git@hotfix/0.9.6#egg=nuitka`:

```
- Name: My App Build
  uses: Nuitka/Nuitka-Action@v0.5
  with:
    nuitka-version: @hotfix/0.9.6
    script-name: my_app.py
    standalone: true
```

**Comment by kayhayen** [comment]
The `hotfix/` branches are ones that go away, once I think of it, do they are probably bad to use in an example. Maybe just use the resulting tag, that would strip the `hotfix` part. 

I just noticed, I have not an example, of how to install a tag anywhere in my Nuitka, while I think I am kind of recommending to use them for Nuitka commercial at least. I could not make curl work on https, there is something that makes this only work with git protocol, I assume. But since branche names are commit-ish, does it really need a separate case then.

So does the distinction make sense? The URLs allow all of it. I agree it will be nice, if you kind of create that URL for them, esp. as using a wrong URL can get you into trouble that gives no good error.

**Comment by xinetzone** [comment]
Consider switching to a friendlier strategy:
![github pages](https://user-images.githubusercontent.com/30319120/215917019-30e57237-ea91-45e1-a4ed-da657fbdb550.png)

**Comment by kayhayen** [comment]
I intend to fix this later this week. I don't think URLs make sense, instead it will be as you have it in the code now, token present -> commercial repo, otherwise standard repo. I do not want to encourage forks of Nuitka as much with the Action, and URLs are hard to get right.

Then, if it starts with not a digit, I will create the URL for checking out the branch latest, and if it's a digit, a tag. I think these are different.

The default value of version will then be `main` so people follow Nuitka stable releases by default. Right now the default is 0.9.4 or something, I changed it to 1.3.8 but no release contains this yet. I think we need a workflow to produce a new release, once triggered, but I am not too familiar with that, there are examples though, I think I saw something the other day.

That way making a release will be easier for us.

**Comment by kayhayen** [comment]
I did what I said there.

Interestingly, "bash" is usable on the Windows runner. So I merged the install dependencies and changed it to always install from the repo. Checking with one of your reports @jimkring that I privately forked and made work. I then made a 0.5 release, and I think, now that the docs suggest using `@main` we need to protect that branch, such that we can enforce using PRs maybe, or we need to be really careful when changing it. Not sure yet, for now I like the agility.

For the options command build, I would like to be able to also merge this in the future. Using bash for all 3 OSes should make things automatically aligned.
